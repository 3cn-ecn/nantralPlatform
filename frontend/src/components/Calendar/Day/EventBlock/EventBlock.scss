.imageEventBlur {
  width: inherit;
  height: inherit;
  backdrop-filter: blur(3px) brightness(75%);
  display: flex;
  justify-content: center;
  align-items: center;
}

.eventBlockTitleDisplay {
  overflow: hidden;
  overflow-wrap: break-word;
  display: -webkit-box;
   -webkit-line-clamp: 1; // Allow the title on 1 line maximum
   -webkit-box-orient: vertical;
}