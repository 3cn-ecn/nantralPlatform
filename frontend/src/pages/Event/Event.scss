@import './../../index.scss';


// .ListButton {
//   background-color: $primary-red-color;;
//   width: 100px;
//   height: 20px;
// }


// .CalendarButton {
//   background-color: $primary-red-color;;
//   width: 100px;
//   height: 20px;
// }

