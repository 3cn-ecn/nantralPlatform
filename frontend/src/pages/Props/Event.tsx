export interface EventProps {
  color: string | null; // Color of the event card
  date: Date; // Event begin date
  description: string; // Description of the event in html
  get_absolute_url: string; // url of the event from root url
  get_group_name: string; // name of the organiser
  group: string; // slug of the organiser
  image: string | null; // uri of the banner
  is_member: boolean;
  is_participating: boolean; // whether connected user particpate at this event
  location: string; // where the event is taking place
  number_of_participants: number;
  publicity: 'Pub';
  slug: string; // slug of the event
  title: string;
  max_participant: number | null; // number max of participant
  end_inscription: Date | null; // date of the end of inscription
  begin_inscripion: Date | null; // date of the beginning of inscription
  end_date: Date | null; // end date of the event
  ticketing: string | null; // url of inscription form
  effectiveSize?: number;
  globalSize?: number;
}
