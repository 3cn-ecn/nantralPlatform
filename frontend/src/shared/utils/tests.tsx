import React from 'react';
import { QueryClient, QueryClientProvider } from 'react-query';

import { render } from '@testing-library/react';

import { CustomThemeProvider } from '#shared/context/CustomTheme.context';
import { ToastProvider } from '#shared/context/Toast.context';

import '../i18n/config';

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
      cacheTime: 0,
      suspense: true,
      useErrorBoundary(_, query) {
        return query.state.data === undefined;
      },
    },
  },
});

/**
 * Customise the `render` function from '@testing-library/react' but with some
 * default providers so that elements can be rendered correctly:
 * -> language: 'en' by default
 * -> theme: 'light' by default
 *
 * @param element - the element to render
 * @returns
 */
export function renderWithProviders(element: React.ReactElement) {
  return render(
    <CustomThemeProvider>
      <QueryClientProvider client={queryClient}>
        <ToastProvider>{element}</ToastProvider>
      </QueryClientProvider>
    </CustomThemeProvider>
  );
}
