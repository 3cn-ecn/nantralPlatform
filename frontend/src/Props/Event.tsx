import { snakeToCamelCase } from '../utils/camel';

export interface EventProps {
  id: number;
  color: string | null; // Color of the event card
  beginDate: Date; // Event begin date in Date format
  endDate: Date; // Event begin date in Date format
  description: string; // Description of the event in html
  getAbsoluteUrl: string; // url of the event from root url
  groupSlug: string; // slug of the organiser
  groupName: string; // name of the organiser
  image: string | null; // uri of the banner
  isMember: boolean;
  isParticipating: boolean; // whether connected user particpate at this event
  location: string; // where the event is taking place
  numberOfParticipants: number;
  publicity: 'Pub';
  slug: string; // slug of the event
  title: string;
  maxParticipant: number | null; // number max of participant
  endInscription: string | null; // date of the end of inscription
  beginInscription: string | null; // date of the beginning of inscription
  formUrl: string | null; // url of inscription form
  isFavorite: boolean;
  effectiveSize?: number; // The number of horizontal parts the events will used
  globalSize?: number; // The number of parts available
  position?: number; // The position in number of parts in the calendar
  placed?: boolean; // whether the event has been placed in the calendar
}

export function eventsToCamelCase(events: Array<any>) {
  events.forEach((event) => {
    // delete when date update to beginDate
    event.begin_date = event.date;

    // delete when endDate defined forEach event
    if (event.end_date === null) {
      event.end_date = new Date(new Date(event.date).getTime() + 3600000);
    }

    snakeToCamelCase(event, { beginDate: 'Date', endDate: 'Date' });
  });
}
