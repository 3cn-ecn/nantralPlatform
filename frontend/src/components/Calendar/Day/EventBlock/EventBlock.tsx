import React from 'react';
import { Button } from '@mui/material';
import { EventProps } from '../../../../Props/Event';

/**
 * The EventBlock component, which is an event in the calendar display.
 * @param day The number of the day in the week.
 * @param event The event object with the event data.
 * @returns The button to display, already placed and resized.
 */
export function EventBlock(props: {
  day: number;
  event: EventProps;
}): JSX.Element {
  const { day, event } = props;
  let todayBegin = false;
  let startTime = 24;

  // Set the time when the event begins in the day.
  if (event.beginDate.getDay() === day % 7) {
    todayBegin = true;
    if (
      event.beginDate.getHours() !== 0 ||
      event.beginDate.getMinutes() !== 0 ||
      event.beginDate.getSeconds() !== 0
    ) {
      startTime =
        23 -
        event.beginDate.getHours() +
        (59 - event.beginDate.getMinutes()) / 60 +
        (60 - event.beginDate.getSeconds()) / 3600;
    }
  }

  // Set the duration of the event.
  let duration: number;
  if (todayBegin) {
    duration = (event.endDate.getTime() - event.beginDate.getTime()) / 3600000;
  } else if (event.endDate.getDay() === day % 7) {
    duration =
      event.endDate.getHours() +
      event.endDate.getMinutes() / 60 +
      event.endDate.getSeconds() / 3600;
  } else {
    duration = 24;
  }
  return (
    <Button
      variant="contained"
      fullWidth
      onClick={() => {
        console.log(event);
      }}
      style={{
        minWidth: `1px`,
        height: `${Math.min(duration, startTime) * 20}px`,
        padding: '5px',
      }}
    >
      {event.title[0]}
    </Button>
  );
}
