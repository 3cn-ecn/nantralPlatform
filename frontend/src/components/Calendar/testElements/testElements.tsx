import { EventProps } from '../../../Props/Event';

export function createTestEvent() {
  const event: EventProps = {
    color: 'blue',
    beginDate: new Date('2023-02-17T03:24:00'),
    endDate: new Date('2023-02-17T04:24:00'),
    description: 'This is an event to test',
    getAbsoluteUrl: 'https://nantral-platform.fr/account/login/?next=/',
    getGroupName: 'Moi-mÃªme',
    group: 'Mon groupe',
    image: null,
    isMember: true,
    isParticipating: true,
    location: 'My home',
    numberOfParticipants: 15,
    publicity: 'Pub',
    slug: 'The slug',
    title: 'The test event',
    maxParticipant: null,
    endInscription: null,
    beginInscription: null,
    ticketing: null,
    isFavorite: true,
    effectiveSize: 1,
    globalSize: 1,
    position: 0,
    placed: false,
  };

  return event;
}
