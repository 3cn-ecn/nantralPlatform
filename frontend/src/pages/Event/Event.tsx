/* eslint-disable prettier/prettier */
import React from 'react';
import { Box, Tab } from '@mui/material';
import { TabContext, TabList, TabPanel } from '@mui/lab';
import './Event.scss';
import axios from 'axios';

import {
  CalendarMonth,
  CalendarToday,
  CalendarViewDay,
} from '@mui/icons-material';
import { EventProps, eventsToCamelCase } from '../../Props/Event';
import FilterBar from '../../components/FilterBar/FilterBar';
import Calendar from '../../components/Calendar/Calendar';
import Formular from '../../components/Formular/Formular';

/**
 * Event Page, with Welcome message, next events, etc...
 * @returns Event page component
 */
function EventList(props: { events: any }) {
  const { events } = props;
  console.log(events);

  return <p>Ceci est une liste.</p>;
}

function EventCalendar(props: { events: any }) {
  const { events } = props;
  return (
    <>
      <p>Ceci est un calendrier.</p>
      <Calendar events={events}></Calendar>
    </>
  );
}

function EventView(props: { events: any }) {
  const { events } = props;
  const [value, setValue] = React.useState('1');
  const handleChange = (event: React.SyntheticEvent, newValue: string) => {
    setValue(newValue);
  };
  return (
    <TabContext value={value}>
      <Box sx={{ borderBottom: 1, borderColor: 'divider' }}>
        <TabList onChange={handleChange} aria-label="lab API tabs example">
          <Tab label="Liste" value="1" />
          <Tab label="Calendrier" value="2" />
        </TabList>
      </Box>
      <TabPanel value="1">
        <EventList events={events}></EventList>
      </TabPanel>
      <TabPanel value="2">
        <EventCalendar events={events}></EventCalendar>
        <CalendarMonth></CalendarMonth>
        <CalendarViewDay></CalendarViewDay>
        <CalendarToday></CalendarToday>
      </TabPanel>
    </TabContext>
  );
}

function Event() {
  const [events, setEvents] = React.useState<Array<EventProps>>([]);

  React.useEffect(() => {
    axios.get('/api/event').then((res: any) => {
      eventsToCamelCase(res.data);
      setEvents(res.data);
    });
  }, []);

  return (
    <>
      <h1>Évènements</h1>
      <p>Ceci est la page des events</p>
      <div style={{ display: 'flex', justifyContent: 'flex-end' }}>
        <Formular />
        <FilterBar />
      </div>
      <EventView events={events} />
    </>
  );
}

export default Event;
