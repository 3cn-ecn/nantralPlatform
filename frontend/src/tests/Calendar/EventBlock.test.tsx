import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import { EventBlock } from '#components/Calendar/Day/EventBlock/EventBlock';
import { EventProps } from '#types/Event';
import { wrapAndRender } from '#utils/tests';

import { createTestEvent } from './testElements/testElements';

const eventInDay: EventProps = createTestEvent(
  'The slug',
  'The test event',
  new Date('2023-02-17T03:24:00'),
  new Date('2023-02-17T04:24:00')
);
const eventBeginInDay: EventProps = createTestEvent(
  'The slug',
  'The test event',
  new Date('2023-02-17T03:00:00'),
  new Date('2023-03-18T04:24:00')
);
const eventEndInDay: EventProps = createTestEvent(
  'The slug',
  'The test event',
  new Date('2023-02-16T03:24:00'),
  new Date('2023-02-17T04:00:00')
);
const eventMiddleInDay: EventProps = createTestEvent(
  'The slug',
  'The test event',
  new Date('2023-02-16T03:24:00'),
  new Date('2023-02-18T04:24:00')
);
const displayDayEventMiddleInDay = new Date(eventMiddleInDay.startDate);
displayDayEventMiddleInDay.setDate(eventMiddleInDay.startDate.getDate() + 1);

describe('<EventBlock />', () => {
  it('should display an EventBlock in the day', async () => {
    const component = wrapAndRender(
      <BrowserRouter>
        <EventBlock day={eventInDay.startDate} event={eventInDay} />
      </BrowserRouter>
    );
    expect(component.getByRole('button').style.height).toBe('1.2rem');
    expect(component.getByRole('button').style.padding).toBe('0px');
    expect(component.getByRole('button').style.backgroundSize).toBe('cover');
    expect(component.getByRole('button').style.backgroundPosition).toBe(
      'center'
    );
    expect(component.getByRole('button').style.backgroundRepeat).toBe(
      'no-repeat'
    );
    expect(component).toMatchSnapshot();
  });

  it('should display an EventBlock in the end of the day', async () => {
    const component = wrapAndRender(
      <BrowserRouter>
        <EventBlock day={eventBeginInDay.startDate} event={eventBeginInDay} />
      </BrowserRouter>
    );
    expect(component.getByRole('button').style.height).toBe('25.2rem');
    expect(component.getByRole('button').style.padding).toBe('0px');
    expect(component.getByRole('button').style.backgroundSize).toBe('cover');
    expect(component.getByRole('button').style.backgroundPosition).toBe(
      'center'
    );
    expect(component.getByRole('button').style.backgroundRepeat).toBe(
      'no-repeat'
    );
    expect(component).toMatchSnapshot();
  });

  it('should display an EventBlock in the begin the day', async () => {
    const component = wrapAndRender(
      <BrowserRouter>
        <EventBlock day={eventEndInDay.endDate} event={eventEndInDay} />
      </BrowserRouter>
    );
    expect(component.getByRole('button').style.height).toBe('4.8rem');
    expect(component.getByRole('button').style.padding).toBe('0px');
    expect(component.getByRole('button').style.backgroundSize).toBe('cover');
    expect(component.getByRole('button').style.backgroundPosition).toBe(
      'center'
    );
    expect(component.getByRole('button').style.backgroundRepeat).toBe(
      'no-repeat'
    );
    expect(component).toMatchSnapshot();
  });

  it('should display an EventBlock in the all day', async () => {
    const component = wrapAndRender(
      <BrowserRouter>
        <EventBlock day={displayDayEventMiddleInDay} event={eventMiddleInDay} />
      </BrowserRouter>
    );
    // expect(component.getByRole('button').style.height).toBe('480px');
    expect(component.getByRole('button').style.padding).toBe('0px');
    expect(component.getByRole('button').style.backgroundSize).toBe('cover');
    expect(component.getByRole('button').style.backgroundPosition).toBe(
      'center'
    );
    expect(component.getByRole('button').style.backgroundRepeat).toBe(
      'no-repeat'
    );
    expect(component).toMatchSnapshot();
  });
});
