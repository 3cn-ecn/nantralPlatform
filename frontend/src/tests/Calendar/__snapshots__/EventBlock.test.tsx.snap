// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventBlock /> should display an EventBlock in the all day 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth eventButton css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
        style="min-width: 1px; height: 28.799999999999997rem; padding: 0px; background: url(null); background-size: cover; background-position: center; background-repeat: no-repeat; vertical-align: top; text-transform: none;"
        tabindex="0"
        type="button"
      >
        <div
          class="imageEventBlur"
        >
          <div
            class="eventBlockTitleDisplay"
          >
            The test event
          </div>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth eventButton css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
      style="min-width: 1px; height: 28.799999999999997rem; padding: 0px; background: url(null); background-size: cover; background-position: center; background-repeat: no-repeat; vertical-align: top; text-transform: none;"
      tabindex="0"
      type="button"
    >
      <div
        class="imageEventBlur"
      >
        <div
          class="eventBlockTitleDisplay"
        >
          The test event
        </div>
      </div>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<EventBlock /> should display an EventBlock in the begin the day 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth eventButton css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
        style="min-width: 1px; height: 4.8rem; padding: 0px; background: url(null); background-size: cover; background-position: center; background-repeat: no-repeat; vertical-align: top; text-transform: none;"
        tabindex="0"
        type="button"
      >
        <div
          class="imageEventBlur"
        >
          <div
            class="eventBlockTitleDisplay"
          >
            The test event
          </div>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth eventButton css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
      style="min-width: 1px; height: 4.8rem; padding: 0px; background: url(null); background-size: cover; background-position: center; background-repeat: no-repeat; vertical-align: top; text-transform: none;"
      tabindex="0"
      type="button"
    >
      <div
        class="imageEventBlur"
      >
        <div
          class="eventBlockTitleDisplay"
        >
          The test event
        </div>
      </div>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<EventBlock /> should display an EventBlock in the day 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth eventButton css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
        style="min-width: 1px; height: 1.2rem; padding: 0px; background: url(null); background-size: cover; background-position: center; background-repeat: no-repeat; vertical-align: top; text-transform: none;"
        tabindex="0"
        type="button"
      >
        <div
          class="imageEventBlur"
        >
          <div
            class="eventBlockTitleDisplay"
          >
            The test event
          </div>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth eventButton css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
      style="min-width: 1px; height: 1.2rem; padding: 0px; background: url(null); background-size: cover; background-position: center; background-repeat: no-repeat; vertical-align: top; text-transform: none;"
      tabindex="0"
      type="button"
    >
      <div
        class="imageEventBlur"
      >
        <div
          class="eventBlockTitleDisplay"
        >
          The test event
        </div>
      </div>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<EventBlock /> should display an EventBlock in the end of the day 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth eventButton css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
        style="min-width: 1px; height: 25.2rem; padding: 0px; background: url(null); background-size: cover; background-position: center; background-repeat: no-repeat; vertical-align: top; text-transform: none;"
        tabindex="0"
        type="button"
      >
        <div
          class="imageEventBlur"
        >
          <div
            class="eventBlockTitleDisplay"
          >
            The test event
          </div>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth eventButton css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
      style="min-width: 1px; height: 25.2rem; padding: 0px; background: url(null); background-size: cover; background-position: center; background-repeat: no-repeat; vertical-align: top; text-transform: none;"
      tabindex="0"
      type="button"
    >
      <div
        class="imageEventBlur"
      >
        <div
          class="eventBlockTitleDisplay"
        >
          The test event
        </div>
      </div>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
